code/modules/mob/living/carbon/xenomorph/castes/warrior/abilities_warrior.dm

```diff
@@ -93,22 +93,29 @@
 	return TRUE
 
 /datum/action/xeno_action/activable/lunge/use_ability(atom/A)
-	var/mob/living/carbon/xenomorph/warrior/X = owner
+	var/mob/living/carbon/xenomorph/X = owner
 
 	GLOB.round_statistics.warrior_lunges++
 	SSblackbox.record_feedback("tally", "round_statistics", 1, "warrior_lunges")
 	X.visible_message(span_xenowarning("\The [X] lunges towards [A]!"), \
 	span_xenowarning("We lunge at [A]!"))
 
 	X.add_filter("warrior_lunge", 2, gauss_blur_filter(3))
-
 	lunge_target = A
+	//Select grapple toss.
+	var/datum/action/xeno_action/activable/toss/grapple_toss = X.actions_by_path[/datum/action/xeno_action/activable/toss]
+	grapple_toss.action_activate()
```

Comment:
need a null check, what if you are giving lunge to another xeno.

grapple_toss?.action_activate